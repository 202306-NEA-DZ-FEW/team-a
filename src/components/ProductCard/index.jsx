import Image from "next/image";
import { MdShareLocation } from "react-icons/md";

import { getLocationName, truncateString } from "@/lib/helpers";

function ProductCard({ title, location, listingType, imageUrl, category }) {
  const locationName = getLocationName(location);
  const truncatedCategory = truncateString(category, 15);
  const truncateState = truncateString(locationName, 6);

  return (
    <div className='card rounded-3xl group'>
      <figure className='relative h-64 w-52 rounded-3xl'>
        <Image
          src={imageUrl}
          alt={title}
          width={200}
          height={200}
          priority
          sizes='(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw'
          className='group-hover:scale-110 saturate-100 group-hover:blur-[2px] hover:saturate-50 brightness-[0.73] object-cover w-full h-full duration-500'
        />
      </figure>
      <div className='absolute bottom-[5%] flex flex-col px-3 w-52'>
        <h3 className='text-white text-lg font-bold'>{title}</h3>
        <span className='text-sm text-white pl-1 font-light mb-1'>
          {truncatedCategory}
        </span>
        <div className='flex items-center gap-1 p-0 w-full'>
          <span className='badge badge-md font-light flex gap-1 text-white border-opacity-30 bg-black bg-opacity-25'>
            {truncateState}
            <span>
              <MdShareLocation />
            </span>
          </span>
          <span className='badge badge-md font-light text-white border-white border-opacity-30 bg-black bg-opacity-25'>
            {listingType}
          </span>
        </div>
      </div>
    </div>
  );
}

export default ProductCard;
