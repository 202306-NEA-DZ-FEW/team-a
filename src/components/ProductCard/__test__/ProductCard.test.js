import image from "public/images/mepi1.png";
import renderer from "react-test-renderer";

import ProductCard from "..";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ProductCard
        title='title'
        location='location'
        listingType='listingType'
        imageUrl={image}
        category='category'
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
