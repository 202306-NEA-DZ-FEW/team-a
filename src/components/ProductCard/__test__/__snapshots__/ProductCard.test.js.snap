// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card w-52 bg-white shadow-xl"
>
  <img
    alt="Shoes"
    className="h-44 p-1 object-cover rounded-tr-2xl rounded-tl-2xl"
    data-nimg="1"
    decoding="async"
    height={400}
    onError={[Function]}
    onLoad={[Function]}
    src=""
    style={
      Object {
        "color": "transparent",
      }
    }
    width={400}
  />
  <div
    className="p-3 flex flex-col gap-1"
  >
    <div
      className="flex items-center"
    >
      <svg
        className="text-2xl text-error"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5a2.5 2.5 0 010-5 2.5 2.5 0 010 5z"
        />
      </svg>
      <span
        className="text-md font-light"
      />
    </div>
    <h2
      className="text-lg font-bold "
    />
    <p
      className="text-sm font-light"
    />
  </div>
</div>
`;
